db/hints/manager.cc

```diff
@@ -50,9 +50,10 @@ const std::string manager::FILENAME_PREFIX("HintsLog" + commitlog::descriptor::S
 const std::chrono::seconds manager::hint_file_write_timeout = std::chrono::seconds(2);
 const std::chrono::seconds manager::hints_flush_period = std::chrono::seconds(10);
 
-manager::manager(sstring hints_directory, host_filter filter, int64_t max_hint_window_ms, resource_manager& res_manager, distributed<database>& db)
+manager::manager(sstring hints_directory, const utils::updateable_value<host_filter>& filter, int64_t max_hint_window_ms, resource_manager& res_manager, distributed<database>& db)
     : _hints_dir(fs::path(hints_directory) / format("{:d}", this_shard_id()))
-    , _host_filter(std::move(filter))
+    , _host_filter(filter())
```

Comment:
Unless I miss something given the current implementation this is going to return a "disable all" filter for view hints manager instantiation (goes through the default constructor).

If I'm right then there is a question if you tested that this series works for both regular and view hints?